spring-xd-dirt/src/main/java/org/springframework/xd/dirt/module/support/ModuleDefinitionService.java

```diff
@@ -86,6 +98,50 @@ public ModuleDefinition findDefinition(String name, ModuleType type) {
 	}
 
 	public ModuleDefinition compose(String name, ModuleType typeHint, String dslDefinition, boolean force) {
+		ModuleDefinition moduleDefinition = null;
+		if(typeHint == ModuleType.job){
+			moduleDefinition = composeJob(name, typeHint, dslDefinition, force);
+		}else{
+			moduleDefinition = composeStream(name, typeHint, dslDefinition, force);
+		}
+		return moduleDefinition;
+	}
+
+	private ModuleDefinition composeJob(String name, ModuleType typeHint, String dslDefinition, boolean force){
+		assertModuleUpdatability(name, typeHint, force);
+		String composedJobXml = composedJobParser.parse(dslDefinition).toXML(name);
+		byte bytes[] = createComposedJobJar(composedJobXml);
+		ModuleDefinition moduleDefinition = new UploadedModuleDefinition(name, typeHint, bytes) ;
+		Assert.isTrue(this.registry.registerNew(moduleDefinition), moduleDefinition + " could not be saved");
+		return moduleDefinition;
+	}
+
+	private byte[] createComposedJobJar(String xml) {
+
+		try (ByteArrayOutputStream outStream = new ByteArrayOutputStream(); JarOutputStream target = new JarOutputStream(outStream)) {
+			JarEntry entry = new JarEntry("config/");
+			target.putNextEntry(entry);
+			target.closeEntry();
+			writeXML(target, xml);
+			target.close();
+			return outStream.toByteArray();
+		} catch (IOException ioe) {
+			throw new IllegalStateException(ioe.getMessage(), ioe);
+		}
+	}
+
+	private void writeXML(JarOutputStream target, String xml) {
+		try (InputStream in = new ByteArrayInputStream(xml.getBytes(Charset.forName("UTF-8")))) {
```

Comment:
Sorry...my mistake.  I forgot that this was introduced in java 7 and not java 8.  You should be fine here (crawling back under my rock now).